crates/bevy_debug_draw/src/debug_draw.rs

```diff
@@ -0,0 +1,200 @@
+use std::f32::consts::TAU;
+
+use bevy_asset::Handle;
+use bevy_ecs::system::Resource;
+use bevy_math::{vec3, Quat, Vec2, Vec3};
+use bevy_render::prelude::{Color, Mesh};
+
+/// A resource with an immediate mode API for drawing lines and wireshapes.
+/// Useful for visual debugging.
+#[derive(Resource)]
+pub struct DebugDraw {
+    positions: Vec<[f32; 3]>,
+    colors: Vec<[f32; 4]>,
```

Comment:
Almost tempted to ask this to be put into a `Mutex` to allow multiple systems to draw at the same time. Otherwise, a `ResMut` on this will force all debug-enabled systems to serialize.

Reply:
Note that <https://github.com/nicopap/bevy-debug-text-overlay/> does just this. Though it uses a mpsc sync_channel, and has a system that reads the receiver once per frame. 

Another advantage of such a method is that you don't need to specify the `DebugLines` resource on the systems that use it

Reply:
I'll be testing both approaches to see how the performance and ergonomics compare :)